packages/nest/src/exclusive-field.validator.ts

```diff
@@ -0,0 +1,102 @@
+import { registerDecorator, ValidationArguments } from 'class-validator';
+
+/**
+ * Decorator to ensure that only one of the fields in a class exists, or one
+ * of the fields in a select group of fields in the class exists.
+ *
+ * @param fields - An optional array of fields or an array of an array of fields
+ * of which each grouping is mutually exclusive. If any of the fields in
+ * each group are present, the others must be null or undefined.
+ *
+ * Given the following class:
+ *
+ * class CharacterFealty {
+ *     orcs?: boolean;
+ *     elves?: boolean;
+ *     humans?: boolean;
+ *     dwarves?: boolean;
+ *   }
+ *
+ * @example
+ * @ExclusiveFieldExists()
+ * fealty: CharacterFealty;
+ *
+ * The existence of any of the fields in CharacterFealty is mutually exclusive.
+ *
+ * @example
+ * @ExclusiveFieldExists(['orcs', 'elves'])
+ * fealty: CharacterFealty;
+ *
+ * The existence of 'orcs' and 'elves is mutually exclusive
+ *
+ * @example
+ * @ExclusiveFieldExists([['orcs', 'elves'],['humans', 'dwarves']])
+ * fealty: CharacterFealty;
+ *
+ * The existence of 'orcs' and 'elves is mutually exclusive, as is the existence
+ * of 'humans' and 'dwarves'. However, 'orcs' and 'humans' can exist together.
+ */
+export const ExclusiveFieldExists =
+  (fields?: string[] | string[][]): PropertyDecorator =>
+  (object: any, propertyName: string | symbol) => {
+    registerDecorator({
+      name: 'exclusiveFieldExists',
+      target: object.constructor,
+      propertyName: propertyName as string,
+      constraints: fields as any[],
+      validator: {
+        validate: (val, args: ValidationArguments) => {
+          const { value } = args;
+          let { constraints } = args;
+          const suppliedKeys = Object.entries(value).flatMap(([key, val]) =>
+            val != null ? key : [],
+          );
+          // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
```

Comment:
@CarsonF - Happy to know of a way to eliminate this `eslint-disable`.

Reply:
I wouldn't bother with using `constraints`. I don't really get their use case. As far as I can tell they are only to be passed through to `validate` and `defaultMessage`.
We can just use `fields` directly.

Reply:
Sorry, I don't understand how to access the `[orcs, elves]` array from `@ExclusiveFieldExists(['orcs', 'elves'])` within the `validate` method unless I use `constraints`?

Reply:
`fields` are scoped into this entire block, so they can just be used directly.
Simply:
```diff
- if (constraints) {
+ if (fields) {
```